bpf/lib/egress_gateway.h

```diff
@@ -52,7 +52,8 @@ bool egress_gw_request_needs_redirect(struct iphdr *ip4 __maybe_unused,
 	if (ct_status == CT_REPLY || ct_status == CT_RELATED)
 		return false;
 
-	egress_gw_policy = lookup_ip4_egress_gw_policy(ip4->saddr, ip4->daddr);
+	egress_gw_policy = lookup_ip4_egress_gw_policy(ipv4_ct_reverse_tuple_saddr(rtuple),
+						       ipv4_ct_reverse_tuple_daddr(rtuple));
```

Comment:
Ah nice, with the preceeding checks we can even be certain that the tuple is in forward-direction layout.

Reply:
I was thinking about making these helpers work with both directions and dynamically return the correct or swapped address based on the `TUPLE_F_{IN,OUT}` value of `flags`, but as I was not sure about complexity implications I decided to go with this simpler thing for now that just helps with readability (and experiment with that in a separate pr)

Reply:
Yep, experimenting sounds good :+1:.

We've started using specific terminology when reviewing CT PRs recently ("NAT layout" for tuples that match the packet headers, and "CT layout" for tuples that have flipped ports). Maybe it's time to bake this into actual helper names etc..